### PR TITLE
Making sure PBS environment uses specified node list even if no PBS_NODEFILE env is available

### PR DESCRIPTION
**Pull request by hkaiser**
@quantomb: pls verify whether this fixes the PBS issues on supermic (when running with TCP).

Generally, the PBS integration in HPX looks only at the `PBS_NODENUM` and `PBS_NODEFILE` (if available) environment variables. It is possible to overwrite the node list using the `--hpx:nodes=node1 node2...` command line option).

All other parameters (like number of threads to use etc.) have to be passed on the command line.

### CODE REVIEW

**Comment by quantomb** [comment]
on it
-Samuel

File: hpx/util/batch_environments/pbs_environment.hpp [hunk 1]
```diff
@@ -53,7 +31,7 @@ namespace hpx { namespace util { namespace batch_environments {
 
         std::size_t num_threads() const
         {
-            return num_threads_;
+            return std::size_t(-1);
```
Comment (sithhell) [inline hpx/util/batch_environments/pbs_environment.hpp]: Why did you remove the computation of the number of threads?

Comment (hkaiser) [inline hpx/util/batch_environments/pbs_environment.hpp]: I removed it because we were handling the list of nodes provided as a nodefile differently from handling a list of nodes given on the command line. The further interpreted the number of times a node was given as the number of threads to use (which - I think - is questionable), the latter was not doing this at all. I'd rather either let the user explicitly specify the number of threads to run on or to look at the `PBS_NUM_VNODES` environment variable (or similar) to extract that number from PBS.

Comment (sithhell) [inline hpx/util/batch_environments/pbs_environment.hpp]: That's a breaking change, at least, I think we should rather change how the given nodelist is being interpreted. If you have a PBS job, and have a mppwidth=24 (that's what you want on supermic for example), you get each host repeated 24 times in the PBS_NODEFILE. To use that, you'd have to do some bash acrobatics to extract the number of nodes.

**Comment by hkaiser** [comment]
@sithhell Your comments have been addressed

**Comment by sithhell** [comment]
LGTM!

**Comment by hendrx** [comment]
Hi Thomas.
woltle nur mal fragen wie das eigtl aussieht mit dem vortrag.
siehe andis email liest sich das jetzt wie nen solo  vortrag von mir. 

in dem fall wuerde ich aber dann eher was zu unserm LZE kram erzaehlen, weil auf der supercomputing war ja das Tutorial das du geplant und organisiert hast jetzt eher so dass ich da mitgeholfen habe, also klar ich kann da deine slides nochmal zeigen usw, aber dann stehl ich dir ja weng die show oder??!

Soll ich dann 30 minuten erzaehlen: "hey subba, ja wir haben halt demos gezeigt und mit leuten geredet ? "

oder willst du da an dem termin ( 9.12 ) auch noch was mit sagen?

cheers,
arne
